src/array/cpu/spmat_op_impl_csr.cc

```diff
@@ -492,7 +492,55 @@ template CSRMatrix CSRSliceRows<kDGLCPU, int64_t>(CSRMatrix, NDArray);
 
 template <DGLDeviceType XPU, typename IdType>
 CSRMatrix CSRSliceMatrix(
-    CSRMatrix csr, runtime::NDArray rows, runtime::NDArray cols) {
+    CSRMatrix csr, runtime::NDArray rows, runtime::NDArray cols,
```

Comment:
For the 2 cases, do we have to create 2 unrelated code block or can we actually share some part of the work?

If we have to create 2 big block, I'd do:
```
CSRMatrix CSRSliceMatrixWithNodesRelabeling() {...}
CSRMatrix CSRSliceMatrixWithOutNodesRelabeling() {...}

CSRMatrix CSRSliceMatrix(bool relabel_nodes) {
  return relabel_nodes ? CSRSliceMatrixWithNodesRelabeling() : CSRSliceMatrixWithOutNodesRelabeling();
}
```

Reply:
OK, I try to make the change in this function simpler.